[DOC] High level docs (#621)
nipoppy/nipoppy: docs/source/overview/principles.md

# Inspirations and principles

> Nipoppy design is inspired by and built upon the previous work of others. These include community standards (e.g. [BIDS](https://bids.neuroimaging.io/)), tools (e.g. [Boutiques](https://boutiques.github.io/)), and technologies (e.g. [Apptainer](https://apptainer.org/)).

Reviewer: ```suggestion
Nipoppy is inspired by and built upon the previous work of others. These include community standards (e.g. [BIDS](https://bids.neuroimaging.io/)), tools (e.g. [Boutiques](https://boutiques.github.io/)), and technologies (e.g. [Apptainer](https://apptainer.org/)).
```

Nipoppy will always strive to support and contribute to existing open-source community standards and avoid creating a new one.
We see Nipoppy as a “dry lab protocol” that glues together existing tools to help adopt open-science and FAIR data principles in practice.

This umbrella effort is guided by following design principles:

| Principle   | Example implementation in Nipoppy    |
|:-------|:------------------------------------------------------------------------|
| Specify sequence / ordering of tasks whenever possible   | Generate a manifest before anything else!                                |
| Optimize for user-oriented modular design    | Conceptually, Nipoppy is divided into *curate, process, extract* modules to match typical research project stages in a neuroimaging lab. |
| Prioritize lightweight, simpler design over comprehensive functionalities  | Nipoppy performs simpler sanity checks at module endpoints instead of sophisticated provenance tracking of the process to ensure reproducibility|
| Handle collected data “as-is” | Nipoppy doesn’t interfere with the current data collection practice i.e. modify recruitment, assessment, DICOM files |

> | Don’t be a black-box | Nipoppy wants to be a hand-on training tool helping researcher to adopt FAIR principles |

Reviewer: ```suggestion
| Don’t be a black box | Nipoppy wants to be a hand-on training tool helping researcher to adopt FAIR principles |
```

| Build for iterative workflows | FAIR research involves **re**producibility, **re**use, and **re**plication. Nipoppy tries to simplify these tasks. |

> | There are no stupid GH issues, or questions on Discord or Neurostar!| Nipoppy is by and for the community - we always welcome feedback to improve it!  |

Reviewer: We don't have a Neurostars tag

```suggestion
| There are no stupid GitHub issues or questions on Discord | Nipoppy is by and for the community - we always welcome feedback to improve it!  |
```

Author: I think this is okay as a general principle sentence. We can also create a tag.

> 

Reviewer 2: I love this :heart_eyes:
